src/v/resource_mgmt/memory_groups.cc

```diff
@@ -33,9 +33,11 @@ struct memory_shares {
     constexpr static size_t recovery = 1;
     constexpr static size_t tiered_storage = 1;
     constexpr static size_t data_transforms = 1;
+    constexpr static size_t datalake = 1;
 
     static size_t total_shares(bool with_wasm) {
-        size_t total = chunk_cache + kafka + rpc + recovery + tiered_storage;
+        size_t total = chunk_cache + kafka + rpc + recovery + tiered_storage
+                       + datalake;
```

Comment:
Also, any color here w.r.t whether to gate this on the iceberg_enabled cluster property?

Reply:
this needs a discussion with Matt. I'm leaving it as is for now.

Reply:
FWIW this does has implications on how much memory subsystems actually use and we should be more strict here for new work, so I would be in favor with respecting the cluster property for modifying the memory for subsystems.

Reply:
right, the problem is toggling it requires a restart of the cluster for the new reservations to take effect which didn't seem like a great UX (especially for someone who just want to try iceberg) .. but agree with the sentiment though.. I had a todo to discuss with Matt about this and fix it with a followup if needed.

Reply:
SG please socialize the discussion in #p-iceberg